File: problems/sorting/lc_2545.py
from typing import List


class Solution:
    def sortTheStudents(self, score: List[List[int]], k: int) -> List[List[int]]:
        score.sort(key=lambda x: x[k], reverse=True)
        return score


# fmt: off
score = [
    [10, 6,  9,  1],
    [ 7, 5, 11,  2],
    [ 4, 8,  3, 15]
]
k = 2

assert Solution().sortTheStudents(score, k) == [
    [ 7, 5, 11,  2],
    [10, 6,  9,  1],
    [ 4, 8,  3, 15],
]
# fmt: on

File: problems/sorting/lc_2545.cpp
#include <algorithm>
#include <vector>
using std::vector;
namespace ranges = std::ranges;

/**
 * std::ranges::sort -> https://en.cppreference.com/w/cpp/algorithm/ranges/sort
 * std::ranges::less -> https://en.cppreference.com/w/cpp/utility/functional/ranges/less
 * std::identity -> https://en.cppreference.com/w/cpp/utility/functional/identity
 */

// clang-format off
class Solution {
public:
  vector<vector<int>> sortTheStudents(vector<vector<int>> &score, int k) {
    ranges::sort(score, {}, [&](auto &row){ return -row[k]; });
    return score;
  }
};
// clang-format on
